src/rmem.c

```diff
@@ -0,0 +1,470 @@
+#include "rmem.h"
+
+
+typedef uintptr_t ptrcmp;
+
+static size_t _AlignSize(const size_t size, const size_t align)
```

Comment:
* `/^_[A-Z]/` identifiers are reserved for the implementation (i.e. compiler and system libraries), don't use them in user code.
* Add an `inline` as an extra hint to the compiler.
* I am not a fan of const parameters. I'd drop them, but that's IMO.